python.d/ceph.chart.py

```diff
@@ -0,0 +1,183 @@
+
+    def create_definitions(self):
+        """
+        Create dynamically charts options, add presentation data/value
+        set on YAML file
+        :return: None
+        """
+        # Options
+        self.definitions['general']['options'] = [None, 'Ceph General Options',
+                                                  'usage ({})'.format(self.presentation_data),
+                                                  'general', 'ceph.general', 'areastack']
+        self.definitions['pool_usage']['options'] = [None, 'Ceph Pools',
+                                                     'usage ({})'.format(self.presentation_data),
+                                                     'pool usage', 'ceph.pool_usage', 'line']
+        self.definitions['osd_usage']['options'] = [None, 'Ceph OSDs',
+                                                     'usage ({})'.format(self.presentation_data),
+                                                     'osd usage', 'ceph.osd_usage', 'line']
+
+        # Lines
+        self.definitions['general']['lines'].append(['general_total', 'Total',
+                                                'absolute', 1, self.presentation_value])
+        self.definitions['general']['lines'].append(['general_usage', 'Usage',
+                                                'absolute', 1, self.presentation_value])
+        self.definitions['general']['lines'].append(['general_available', 'Available',
+                                                'absolute', 1, self.presentation_value])
+
+        for pool in self.cluster.list_pools():
+            self.definitions['pool_usage']['lines'].append([pool, pool,
+                                                            'absolute', 1, self.presentation_value])
+            self.definitions['pool_objects']['lines'].append(["obj_{}".format(pool), pool,
+                                                              'absolute'])
+
+        for osd in self._get_osds_df()['nodes']:
+            self.definitions['osd_usage']['lines'].append([osd['name'], osd['name'],
+                                                           'absolute', 1, self.presentation_value])
+
+    def get_data(self):
+        """
+        Catch all ceph data
+        :return: dict
+        """
+        try:
+            data = {}
+            data.update(self._get_general())
+            data.update(self._get_pool_usage())
+            data.update(self._get_pool_objects())
+            data.update(self._get_osd_usage())
+            return data
+        except (ValueError, AttributeError):
+            return None
+
+    def _get_general(self):
+        """
+        Get ceph's general usage
+        :return: dict
+        """
+        info = self.cluster.get_cluster_stats()
+        # Values are get in bytes, convert it
+        return {'general_total': float(info['kb']),
+                'general_usage': float(info['kb_used']),
+                'general_available': float(info['kb_avail'])
```

Comment:
https://github.com/firehol/netdata/blob/master/python.d/python_modules/bases/charts.py#L312-L316

> Only integer values are collected. If you want to push fractional values, multiply this value by 100 or 1000 and set the DIMENSION divider to 1000.

Reply:
I forgot to remove the float casting. I will fix this.